packages/flutter_tools/lib/src/compile.dart

```diff
@@ -577,7 +599,16 @@ class DefaultResidentCompiler implements ResidentCompiler {
     if (!_controller.hasListener) {
       _controller.stream.listen(_handleCompilationRequest);
     }
-
+    // Write `generated_main.dart` under the tool's owned directory.
+    // This ensures that the `lib` directory isn't polluted with generated files.
+    final Directory buildDir = _fileSystem.file(mainUri.path)
+        .parent
+        .parent
+        .childDirectory(_fileSystem.path.join('.dart_tool', 'flutter_build'));
+    final File newMainDart = buildDir.childFile('generated_main.dart');
+    if (newMainDart.existsSync()) {
```

Comment:
In what cases does this not exist?

Reply:
If there aren't plugins or if no plugin needs a Dart registry